internal/sysfs/file.go

```diff
@@ -38,9 +38,6 @@ func NewStdioFile(stdin bool, f fs.File) (fsapi.File, error) {
 }
 
 func OpenFile(path string, flag experimentalsys.Oflag, perm fs.FileMode) (*os.File, experimentalsys.Errno) {
-	if flag&experimentalsys.O_DIRECTORY != 0 && flag&(experimentalsys.O_WRONLY|experimentalsys.O_RDWR) != 0 {
```

Comment:
This check is safe to remove because the write flag is now properly propagated to the underlying syscall instead of being masked by an early return in `func openFlags`.

Reply:
It is also necessary to remove this for correctness because this check sometimes erroneously return `isdir` when the target is not actually a directory.

Reply:
would you be able to add a (negative) test for when this happens, i.e. a case where `main` still returns `isdir` for something that is not a dir? so we won't erroneously restore the incorrect behavior in the future.